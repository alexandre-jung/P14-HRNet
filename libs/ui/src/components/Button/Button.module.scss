$button-size: 32px;

.Button {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  min-width: $button-size;
  min-height: $button-size;
  background-color: aliceblue;
  border: none;
  padding: 5px 10px;

  &-primary {
    background-color: aqua;
  }

  &-secondary {
    background-color: burlywood;
  }

  &-error {
    background-color: firebrick;
    color: white;
  }
}
