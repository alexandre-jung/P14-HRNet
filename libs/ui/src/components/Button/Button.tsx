import React from 'react';

import styles from './Button.module.scss';
import { ButtonProps } from './Button.types';

export default function Button ({ type, text, onClick }: ButtonProps) {
  const typeClassName = styles[`Button-${type}`];

  return (
    <button
      type="button"
      className={`${styles.Button} ${typeClassName}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
}
