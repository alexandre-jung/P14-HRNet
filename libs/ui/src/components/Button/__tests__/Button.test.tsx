import { render } from '@testing-library/react';
import Button from '..';

describe('Button', () => {
  it('should work', function () {
    const { container } = render(
      <Button type="error" text="Hello world!" onClick={() => {}} />,
    );

    expect(container).toMatchSnapshot();
  });
});
