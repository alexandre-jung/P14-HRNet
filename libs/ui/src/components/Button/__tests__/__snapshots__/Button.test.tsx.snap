// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should work 1`] = `
<div>
  <button
    class="buttonModule.Button buttonModule.Button-error"
    type="button"
  >
    Hello world!
  </button>
</div>
`;
